tests/baselines/reference/api/tsserverlibrary.d.ts

```diff
@@ -7053,22 +7053,32 @@ declare namespace ts.server.protocol {
      * compiler settings.
      */
     type ExternalProjectCompilerOptions = CompilerOptions & CompileOnSaveMixin & WatchOptions;
+    interface FileWithProjectReferenceRedirectInfo {
+        /**
+         * Name of file
+         */
+        fileName: string;
+        /**
+         * True if the file is primarily included in a referenced project
+         */
+        isSourceOfProjectReferenceRedirect: boolean;
+    }
     /**
      * Represents a set of changes that happen in project
      */
     interface ProjectChanges {
         /**
          * List of added files
          */
-        added: string[];
+        added: string[] | FileWithProjectReferenceRedirectInfo[];
```

Comment:
Wait, isn't this a breaking protocol change?  Don't we have to add a parallel array of flags that may or may not be provided?

Reply:
I think the idea is that it would only return the new type if the client asked for it in the request. (Maybe this should be clarified in the doc comment for the property though)

Reply:
Updated the doc comment to clarify this. :)